src/models/Form.js

```diff
@@ -227,11 +227,8 @@ module.exports = (formio) => {
   model.schema.index(hook.alter('schemaIndex', {type: 1, deleted: 1, modified: -1}));
   model.schema.index(hook.alter('schemaIndex', {name: 1, deleted: 1}));
 
-  // Add a partial index for deleted forms.
   model.schema.index({
     deleted: 1
-  }, {
```

Comment:
Is there something stopping us from using try/catch to attempt a partialFilterExpression and to fall back to a normal index if it fails (i.e. if we're in a DocumentDB instance)? 

We do something similar with CosmosDB and TTL indexes [here](https://github.com/formio/formio/blob/252049c1340e43137f60f08219a6b7c88c38208a/src/models/ActionItem.js#L60) - CosmosDB doesn't support expiry params in normal indexes, so we attempt to perform a normal MongoDB API index creation, and if it fails we assume CosmosDB and attempt the CosmosDB API index creation.

Reply:
The idea looks good, but the problem I'm facing is that when you first create the indexes of the collection (in our case Submission and Forms), documentDB dose not return errors, it simple doesn’t create this kind of indexes (delete in our case), so we lose delete indexes for our base tables without possibility to catch this situation. So I decided that the use of the normal index is more reliable here.

Reply:
Aaaah, I tested this a little this morning, and I believe that it's Mongoose that is the culprit, failing silently on index creation (as I assume it **also** wants to be as agnostic as it can be about database vendors). Since we're generally leaning away from Mongoose, your solution is the best option. Nicely done!